### DISCUSSION TITLE
Typescript/Javascript Syntax highlighting stop working after upgrade

### DISCUSSION THREAD

**Discussion by francisl** [helix-editor/helix#8643] [Troubleshooting]
### Summary

after upgrading to 23.10 (from 23.03) on Mac OS with home-brew.
Syntax highlighting for typescript and javascript stop working.

Updating the grammar did the trick. But there is not a clear documentation to help us to do so.

### Reproduction Steps

 

I tried this:
1. Open javascript file
2. Open typescript file
3. open other file (go, html, json)

In combination with:
1. previous languages.toml config
2. Updated from new Doc
3. Empty languages.toml file

only TS and JS file stop working (+tsx, jsx)

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
please provide a copy of `~/.cache/helix/helix.log` here if possible, you may need to redact some of the lines
```

```

2023-10-27T12:44:07.172 helix_core::syntax [ERROR] Could not parse queries for language "tsx". Are your grammars out of sync? Try running 'hx --grammar fetch' and 'hx --grammar build'. This query could not be parsed: QueryError { row: 109, column: 23, offset: 2026, message: "", kind: NodeType }
2023-10-27T12:44:10.022 helix_core::syntax [ERROR] Could not parse queries for language "typescript". Are your grammars out of sync? Try running 'hx --grammar fetch' and 'hx --grammar build'. This query could not be parsed: QueryError { row: 158, column: 3, offset: 3043, message: "satisfies", kind: NodeType }
2023-10-27T12:44:11.257 helix_term::application [ERROR] received malformed notification from Language Server: Unhandled
```

</details>

### Platform

MacOS

### Terminal Emulator

xterm-256color

### Helix Version

23.10

**Comment by the-mikedavis** [comment]
Did you install Helix with brew's head installs? (Those point to master rather than releases and are built on-the-fly.)

Did you ever run `hx -g fetch` and `hx -g build` when you were using 23.03?

**Comment by doneill** [comment]
I can't get the syntax highlighting to work, get errors on health check as well

```
hx --health typescript
Error parsing user language config: invalid type: map, expected a sequence
in `language.language-servers`

Using default language config
Configured language servers:
  ✓ typescript-language-server: /opt/homebrew/bin/typescript-language-server
Configured debug adapter: None
Highlight queries: ✓
Textobject queries: ✓
Indent queries: ✓
```

Is there any resource for converting language.toml files from previous releases?

**Reply by doneill** [reply]
Fixed grammars with `fetch`/`build`

**Comment by gamma-delta** [comment]
Hello, I'm getting this same problem. I've fetched and built all the grammars and I'm on 23.10-1 (most recent version at time of writing).

In JS files I have the LSP working great, and indenting seems to work correctly, but no syntax highlighting. Manually setting the language to typescript makes the syntax highlighting work great.

In the meantime I'm probably just gonna configure .js files to be typescript in Helix, ...

**Reply by kirawi** [reply]
Have you checked `:log-open`?

**Comment by AjBreidenbach** [comment]
Deleting `runtime/` and recreating fixes
